index.html

```diff
@@ -541,11 +541,14 @@ <h1>{{ chosen_domain == 'US' ? 'USA' :
   scale = scale || 'log10';
   if (chart) { chart.detach(); }
   if (stat == 'deltas') {
+    // How many days back do we go to calculate the delta
+    const deltaInterval = 4;
     extract.series = _.map(extract.series, p => [p[0], _.map(p[1], (d, i) => {
       if (i == 0) { return null; }
+      var prevIndex = Math.max(0, i - deltaInterval);
```

Comment:
Instead of pretending days before day 0 are the same as the value on day 0, we should put in the actual value.
To do this properly, we really shouldn't hack the beginning like this when there exist data before day 0.

That means we probably should change the `apply_start_rule` function.  Proposal: that function identifies the threshold day for each series and then drops data before day 0.  Instead, it should pad the arrays (with null or undefined values) to align day 0 in the same place for each series without dropping any data, and then the object it should return should have a `day_zero: i` field that is used later to plot day zero at the start.

Then here we can say if prevIndex < 0, we can assume the value is 0.

Reply:
> Instead of pretending days before day 0 are the same as the value on day 0

To be precise, that's not exactly what it's doing. Instead, it's simply reducing the time window. e.g. if the first two numbers are 3 and 5, it'll end up with 4. You make it sound like it would end up with the average of 3,3,3,...,3,5, which would be very skewed toward the first number. So you're right that it's ignoring real data that comes before the graphing start, but it's doing it in a more reasonable way :)

Alternate fix idea would be to take delta processing out of `plot_series` and move it before `apply_start_rule`, which would make things work naturally without adding extra logic. But that could be disrupting some other things?

Though I'm wondering how much this really matters. If our delta interval is small (like 4), and the graph is going over a large set of days (it usually is), I'm guessing the difference would be barely noticeable between 'reducing the interval size at the edge' (what I have now) and doing the more correct thing by padding. It's about questioning whether the added complexity of keeping track of the `day_zero` index is worthwhile.

Your call, happy to do it either way!

Reply:
After playing with it a bit, I think it's better not to shrink the window, or else we get weird artifacts like the first few cases in california plotted like below, where as the window gets bigger, the number decreases.  I think it's best to just treat "days before the first case" as zero, and "days before the plotted start date" as having their actual value, even though not visible in the plot.

![image](https://user-images.githubusercontent.com/3458792/78859636-f312be00-79fd-11ea-9db6-1c323f2c328a.png)

Ideally we should do it right even if you manually edit the URL to choose a very large window like 30day.  As you noted it requires doing things to avoid interference with other functions, so I've coded it up as a change on top of your pull request.  Will merge soon.